pkg/alertmanager/multitenant.go

```diff
@@ -115,10 +123,17 @@ func (cfg *MultitenantAlertmanagerConfig) RegisterFlags(f *flag.FlagSet) {
 	f.StringVar(&cfg.AutoWebhookRoot, "alertmanager.configs.auto-webhook-root", "", "Root of URL to generate if config is "+autoWebhookURL)
 	f.DurationVar(&cfg.PollInterval, "alertmanager.configs.poll-interval", 15*time.Second, "How frequently to poll Cortex configs")
 
-	f.StringVar(&cfg.ClusterBindAddr, "cluster.listen-address", defaultClusterAddr, "Listen address for cluster.")
-	f.StringVar(&cfg.ClusterAdvertiseAddr, "cluster.advertise-address", "", "Explicit address to advertise in cluster.")
-	f.Var(&cfg.Peers, "cluster.peer", "Initial peers (may be repeated).")
-	f.DurationVar(&cfg.PeerTimeout, "cluster.peer-timeout", time.Second*15, "Time to wait between peers to send notifications.")
+	// Flags prefixed with `cluster` are deprecated in favor of their `alertmanager` prefix equivalent.
+	// TODO: New flags introduced in Cortex 1.7, remove old ones in Cortex 1.9
+	f.StringVar(&cfg.ClusterBindAddr, "cluster.listen-address", defaultClusterAddr, "Deprecated. Use -alertmanager.ha.listen-address instead.")
+	f.StringVar(&cfg.ClusterAdvertiseAddr, "cluster.advertise-address", "", "Deprecated. Use -alertmanager.ha.advertise-address instead.")
+	f.Var(&cfg.Peers, "cluster.peer", "Deprecated. Use -alertmanager.ha.peers instead.")
+	f.DurationVar(&cfg.PeerTimeout, "cluster.peer-timeout", time.Second*15, "Deprecated. Use -alertmanager.ha.peer-timeout instead.")
+
+	f.StringVar(&cfg.HABindAddr, "alertmanager.ha.listen-address", defaultClusterAddr, "Listen address and port for the cluster. Not specifying this flag disables high-availability mode.")
```

Comment:
If `ha` has a dot after it, then the YAML config option should be a sub-struct. Otherwise, keeping the current YAML config as is (my preference) then CLI flags should be name `alertmanager.ha_*`.